File: modules/planning/calendar.py
import os
import json
import datetime

def get_data(config):
    holidays_file = config.get('holidays_file', 'holidays.json')
    today = datetime.date.today()
    results = []
    
    # Holidays
    holiday_str = "No upcoming holidays found"
    try:
        script_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
        json_path = os.path.join(script_dir, holidays_file)
        if os.path.exists(json_path):
            with open(json_path, 'r') as f:
                holidays_data = json.load(f)
                
            upcoming = []
            for date_str, name in holidays_data.items():
                h_date = datetime.datetime.strptime(date_str, "%Y-%m-%d").date()
                if h_date >= today:
                    upcoming.append((h_date, name))
            
            if upcoming:
                upcoming.sort()
                next_h_date, next_h_name = upcoming[0]
                days_away = (next_h_date - today).days
                date_display = next_h_date.strftime("%d %b")
                holiday_str = f"{next_h_name} ({date_display}, {days_away} days)" if days_away > 0 else f"TODAY: {next_h_name}"
        else:
            holiday_str = f"Holidays file not found: {holidays_file}"
    except Exception as e:
        holiday_str = f"Holidays error: {e}"

    results.append(f"Nearest Holiday: {holiday_str}")

    # Sunday
    days_to_sunday = (6 - today.weekday()) % 7
    days_to_sunday = 7 if days_to_sunday == 0 else days_to_sunday
    sunday_date = (today + datetime.timedelta(days=days_to_sunday)).strftime("%d %b")
    results.append(f"Next Sunday: {sunday_date} ({days_to_sunday} days left)")
    
    return results

File: modules/planning/calendar.go
package planning

import (
	"encoding/json"
	"fmt"
	"os"
	"sort"
	"time"
)

type Calendar struct{}

func (Calendar) GetData(cfg map[string]any) []string {
	holidaysFile := "holidays.json"
	if v, ok := cfg["holidays_file"].(string); ok && v != "" {
		holidaysFile = v
	}

	today := time.Now().Truncate(24 * time.Hour)
	var out []string

	holidayStr := "No upcoming holidays found"
	if data, err := os.ReadFile(holidaysFile); err == nil {
		var holidays map[string]string
		if json.Unmarshal(data, &holidays) == nil {
			type entry struct {
				date time.Time
				name string
			}
			var upcoming []entry
			for ds, name := range holidays {
				t, err := time.Parse("2006-01-02", ds)
				if err != nil {
					continue
				}
				if !t.Before(today) {
					upcoming = append(upcoming, entry{t, name})
				}
			}
			sort.Slice(upcoming, func(i, j int) bool {
				return upcoming[i].date.Before(upcoming[j].date)
			})
			if len(upcoming) > 0 {
				next := upcoming[0]
				days := int(next.date.Sub(today).Hours() / 24)
				if days == 0 {
					holidayStr = "TODAY: " + next.name
				} else {
					holidayStr = fmt.Sprintf("%s  (%s, %d days away)",
						next.name, next.date.Format("02 Jan"), days)
				}
			}
		}
	} else {
		holidayStr = fmt.Sprintf("holidays.json not found (%s)", holidaysFile)
	}
	out = append(out, "Nearest Holiday : "+holidayStr)

	weekday := int(today.Weekday())
	daysToSunday := (7 - weekday) % 7
	if daysToSunday == 0 {
		daysToSunday = 7
	}
	nextSunday := today.AddDate(0, 0, daysToSunday)
	out = append(out, fmt.Sprintf("Next Sunday     : %s  (%d days)", nextSunday.Format("02 Jan"), daysToSunday))

	return out
}
